python/cudf/cudf/_libxx/hash.pyx

```diff
@@ -0,0 +1,49 @@
+# Copyright (c) 2020, NVIDIA CORPORATION.
+
+# cython: profile=False
+# distutils: language = c++
+# cython: embedsignature = True
+# cython: language_level = 3
+
+from cudf._libxx.lib cimport *
+from cudf._libxx.column cimport Column
+from cudf._libxx.table cimport Table
+
+from cudf._libxx.includes.hash cimport (
+    hash_partition as cpp_hash_partition,
+    hash as cpp_hash
+)
+
+
+def hash_partition(Table source_table, columns_to_hash, num_partitions):
+    cdef vector[size_type] c_columns_to_hash = columns_to_hash
+    cdef int c_num_partitions = num_partitions
+
+    cdef pair[unique_ptr[table], vector[size_type]] c_result = move(
```

Comment:
We should release the gil before this, basically need to move the `source_table.view()` call above here and then add a:
```
with nogil:
```

Reply:
Is this because the gil needs to be released when calling C functions? (and presumably you can't call Python functions when the gil is released?)

Reply:
Hm, but `source_table.view()` is also a C function, so why can I call that without releasing the gil?

Reply:
`source_table` is a Python class so even though `.view()` is a c function, it requires the GIL to call it.

Reply:
Ok, makes sense